### PR TITLE
TRITON-2462 store affinity rules when provisioning

### PR DESCRIPTION
**Pull request by bahamat**
This accompanies and should be merged *after* TritonDataCenter/sdc-cloudapi#149.

### CODE REVIEW

**Review by danmcd — COMMENTED** [review]
TRITON-779 also had mods for [test/vms.full.test.js](https://github.com/TritonDataCenter/sdc-vmapi/commit/a3b078c4bd6f9d612379122e18e6e35d67c87222#diff-4b39201d0f679c2dea82340f45889651207fc598e56af50d42e7691fc0772b29)  -- do we need corresponding ones here?

File: test/vms.full.test.js [hunk 1]
```diff
@@ -903,7 +904,16 @@ exports.create_vm = function (t) {
         locality: {
             strict: false,
             far: [ '00000000-0000-0000-0000-000000000001' ]
-        }
+        },
+        affinity: [
+            {
+                "key": "instance",
+                "operator": "!=",
+                "value": "00000000-0000-0000-0000-000000000001",
```
Comment (danmcd) [inline test/vms.full.test.js:912]: Not a blocker but I wish to understand:  This rule means "DO NOT MIGRATE ME TO CN UUID == 000....00000001" right?

**Review by danmcd — APPROVED** [review]
One non-blocking question for my own understanding.

Otherwise excellent, just what I had in mind.  Please make sure of copyrights, etc. as well here and in CLOUDAPI.

File: test/vms.full.test.js (hunk 1 above)
Comment (bahamat) [reply test/vms.full.test.js:912]: It means "do not provision to vm with uuid ...", and line 913 shows that it's a soft requirement, so if the constraint can't be satisfied the provision is still allowed anyway.

**Review by danmcd — DISMISSED** [review]

**Review by danmcd — APPROVED** [review]
Shoot, sorry for missing this.

**Pull request merged**